docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,48 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
```

Comment:
For memory it's not just `mstore`. There are many opcodes that read and write memory: https://github.com/ethereum/solidity/blob/8df45f5f8632da4817bc7ceb81497518f298d290/libevmasm/SemanticInformation.cpp#L359-L392

Reply:
Probably the latter. We don't want to have to update the list. It just should not say only `mstore`.